Userland/Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -182,4 +182,52 @@ ThrowCompletionOr<ArrayBuffer*> clone_array_buffer(VM& vm, ArrayBuffer& source_b
     return target_buffer;
 }
 
+// 25.1.2.14 ArrayBufferCopyAndDetach ( arrayBuffer, newLength, preserveResizability ), https://tc39.es/proposal-arraybuffer-transfer/#sec-arraybuffer.prototype.transfertofixedlength
+ThrowCompletionOr<ArrayBuffer*> copy_and_detach(VM& vm, ArrayBuffer& array_buffer, Value new_length, PreserveResizability)
+{
+    auto& realm = *vm.current_realm();
+
+    // 1. Perform ? RequireInternalSlot(arrayBuffer, [[ArrayBufferData]]).
+
+    // FIXME: 2. If IsSharedArrayBuffer(arrayBuffer) is true, throw a TypeError exception.
+
+    // NOTE: This seems more logical to be before step 3, we should probably file a spec bug
```

Comment:
No, the spec generally favours doing observable coercion of input arguments first (ToIndex in this case). Apply the suggestion from the first commit and the issue of accessing byte length goes away :)